dynawo/sources/Modeler/DataInterface/DYNVoltageLevelInterface.h

```diff
@@ -215,6 +215,16 @@ class VoltageLevelInterface {
    * @return @b true if the topology of the voltage level is node breaker one
    */
   virtual bool isNodeBreakerTopology() const = 0;
+
+  /**
+   * @brief Retrieve the buses linked to a bus by a switches network path
+   *
+   * This Service API can be used only if topology is bus breaker
+   *
+   * @param busId bus id to check
+   * @returns the list of the bus ids linked to @p busId
+   */
+  virtual std::vector<std::string> getBusesConnectedBySwitch(const std::string& busId) = 0;
```

Comment:
Before reviewing the algorithm: I'm not sure putting those services inside the VL interface is the best way to do this. This is very specific to DynaFlow, and not used in the core of the simulation. Wouldn't it be possible to have a service manager on the side to perform this?

Reply:
@adriguir any opinion?

Reply:
I think this is a good idea

Reply:
We already have a similar service in the case of node breaker besides this new one. If we create a service manager, we should also move ALL services like this one in it. Seems a lot heavy to me

Reply:
which service are we referring to here? If it is the graph exploration for node braker case it is not really a service as it is required for the core to work